packaging/tests/tas-cmis/src/main/java/org/alfresco/cmis/AuthParameterProviderFactory.java

```diff
@@ -0,0 +1,130 @@
+package org.alfresco.cmis;
+
+import org.alfresco.utility.data.AisToken;
+import org.alfresco.utility.data.auth.DataAIS;
+import org.alfresco.utility.model.UserModel;
+import org.apache.chemistry.opencmis.commons.SessionParameter;
+import org.keycloak.authorization.client.util.HttpResponseException;
+import org.keycloak.representations.AccessTokenResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import static org.alfresco.utility.report.log.Step.STEP;
+
+@Service
+public class AuthParameterProviderFactory
+{
+    public static String STEP_PREFIX = "CMIS AuthParameterProvider:";
+
+    @Autowired
+    private DataAIS dataAIS;
+
+    @Autowired
+    private CmisProperties cmisProperties;
+
+    /**
+     *
+     * The default provider uses AIS if support for Alfresco Identity Service is enabled.
+     * Otherwise a provider which uses Basic authentication is returned.
+     *
+     * @return Function which takes a {@link UserModel} and returns a map of
+     * authentication parameters to be used with {@link CmisWrapper#authenticateUser(UserModel, Function)}
+     */
+    public Function<UserModel, Map<String, String>> getDefaultProvider()
+    {
+        if (dataAIS.isEnabled())
+        {
+            STEP(String.format("%s Retrieved default AIS auth parameter provider.", STEP_PREFIX));
+            return new AisAuthParameterProvider();
+        }
+        else
+        {
+            STEP(String.format("%s Retrieved default Basic auth parameter provider.", STEP_PREFIX));
+            return new BasicAuthParameterProvider();
+        }
+    }
+
+    public Function<UserModel, Map<String, String>> getAISProvider()
+    {
+        return new AisAuthParameterProvider();
+    }
+
+    public Function<UserModel, Map<String, String>> getBasicProvider()
+    {
+        return new BasicAuthParameterProvider();
+    }
+
+    private class BasicAuthParameterProvider implements Function<UserModel, Map<String, String>>

```

Comment:
Implementing `Function` seems _weird_. More idiomatic way would be to use a method reference when returning a function.